test/shell-pipeline-tests/common.sh

```diff
@@ -75,10 +75,93 @@ function createImageRegistrySecret() {
     sed -i "s|{{ IMAGE_REGISTRY_PASSWORD }}|${PASSWORD}|" "$IMAGE_REGISTRY_SECRET_PATH"
 }
 
+function createGitCredentialsSecret() {
+    local GIT_CREDENTIALS_SECRET_PATH_TEMPLATE=$1
+    local GIT_CREDENTIALS_SECRET_PATH=$2
+
+    local GH_TOKEN
+    GH_TOKEN=$(getGitHubToken)
+
+    cp "$GIT_CREDENTIALS_SECRET_PATH_TEMPLATE" "$GIT_CREDENTIALS_SECRET_PATH"
+
+    sed -i "s|{github_pat_1234567890ABCDAPI_TOKEN}|$GH_TOKEN|" "$GIT_CREDENTIALS_SECRET_PATH"
+    sed -i "s|{username}|MarianMacik|" "$GIT_CREDENTIALS_SECRET_PATH"
+}
+
 function usePRBranchInPipelineRunIfPRCheck() {
   local -r PIPELINE_RUN_FILE_PATH=$1
 
   if [ -n "$PULL_NUMBER" ]; then
     sed -i "s|value: \"main\"|value: \"pull/$PULL_NUMBER/head\"|" "$PIPELINE_RUN_FILE_PATH"
   fi
 }
+
+function assertContains() {
+  local -r STRING=$1
+  local -r SUBSTRING=$2
+
+  [[ "$STRING" =~ $SUBSTRING ]] || { echo -e "Assertion failed!\n '$STRING' does not contain '$SUBSTRING'" && return 1; }
+}
+
+assertGitHubPullRequest() {
+  local -r PIPELINE_RUN_NAME=$1
+  local -r PR_URL=$2
+
+  local -r PR_INFO_PATH="pr-info-${PIPELINE_RUN_NAME}.txt"
+  local -r PR_DIFF_PATH="pr-diff-${PIPELINE_RUN_NAME}.txt"
+
+  gh pr view "$PR_URL" > "$PR_INFO_PATH"
+  gh pr diff "$PR_URL" > "$PR_DIFF_PATH"
+
+  # Expected values
+  PIPELINE_RUN_UID=$(oc get pipelinerun "$PIPELINE_RUN_NAME" -o jsonpath={.metadata.uid})
+  FILE_PATH=$(oc get pipelinerun "$PIPELINE_RUN_NAME" -o jsonpath={.spec.params[?\(@.name==\'imageReferenceFilePath\'\)].value})
+  IMAGE_REGISTRY=$(oc get pipelinerun "$PIPELINE_RUN_NAME" -o jsonpath={.status.results[?\(@.name==\'target-registry-url\'\)].value})
+  NEW_DIGEST=$(oc get pipelinerun "$PIPELINE_RUN_NAME" -o jsonpath={.status.results[?\(@.name==\'image-sha\'\)].value})
+  BASE_REF_NAME=$(oc get pipelinerun "$PIPELINE_RUN_NAME" -o jsonpath={.spec.params[?\(@.name==\'gitRepoBranchBase\'\)].value})
+  GIT_SERVER=$(oc get pipelinerun "$PIPELINE_RUN_NAME" -o jsonpath={.spec.params[?\(@.name==\'gitServer\'\)].value})
+  GIT_ORG_NAME=$(oc get pipelinerun "$PIPELINE_RUN_NAME" -o jsonpath={.spec.params[?\(@.name==\'gitOrgName\'\)].value})
+  GIT_REPO_NAME=$(oc get pipelinerun "$PIPELINE_RUN_NAME" -o jsonpath={.spec.params[?\(@.name==\'gitRepoName\'\)].value})
+  GIT_URL="$GIT_SERVER/$GIT_ORG_NAME/$GIT_REPO_NAME"
+
+  # Actual values
+  PR_PIPELINE_RUN_NAME=$(grep "PipelineRun Name" "$PR_INFO_PATH" )
+  PR_PIPELINE_RUN_UID=$(grep "PipelinRun UID" "$PR_INFO_PATH")
+  PR_FILE_PATH=$(grep "File Path" "$PR_INFO_PATH") # TODO Remove
```

Comment:
There's a TODO here. Is that expected to be removed now, or at some future time? If it's in the future, is it something that could/should be tracked in a Jira issue that could be linked here?

Reply:
Yes, it is a reminder that this should be removed after #236 is merged as it removes this field.